File: backend/app/admin/service/internal/service/organization.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-monolithic-demo/app/admin/service/internal/data"

	"kratos-monolithic-demo/gen/api/go/common/pagination"
	v1 "kratos-monolithic-demo/gen/api/go/user/service/v1"
	v12 "kratos-monolithic-demo/gen/api/go/user/service/v1"
)

type OrganizationService struct {
	v12.UnimplementedOrganizationServiceServer

	log *log.Helper

	uc *data.OrganizationRepo
}

func NewOrganizationService(uc *data.OrganizationRepo, logger log.Logger) *OrganizationService {
	l := log.NewHelper(log.With(logger, "module", "organization/service/admin-service"))
	return &OrganizationService{
		log: l,
		uc:  uc,
	}
}

func (s *OrganizationService) ListOrganization(ctx context.Context, req *pagination.PagingRequest) (*v1.ListOrganizationResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *OrganizationService) GetOrganization(ctx context.Context, req *v1.GetOrganizationRequest) (*v1.Organization, error) {
	return s.uc.Get(ctx, req)
}

func (s *OrganizationService) CreateOrganization(ctx context.Context, req *v1.CreateOrganizationRequest) (*v1.Organization, error) {
	return s.uc.Create(ctx, req)
}

func (s *OrganizationService) UpdateOrganization(ctx context.Context, req *v1.UpdateOrganizationRequest) (*v1.Organization, error) {
	return s.uc.Update(ctx, req)
}

func (s *OrganizationService) DeleteOrganization(ctx context.Context, req *v1.DeleteOrganizationRequest) (*emptypb.Empty, error) {
	_, err := s.uc.Delete(ctx, req)
	if err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: frontend/apps/admin/src/rpc/service/organization.ts
import type { OrganizationService } from '#/rpc/api/admin/service/v1/i_organization.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';
import type {
  CreateOrganizationRequest,
  DeleteOrganizationRequest,
  GetOrganizationRequest,
  ListOrganizationResponse,
  Organization,
  UpdateOrganizationRequest,
} from '#/rpc/api/user/service/v1/organization.pb';

import { requestClient } from '#/rpc/request';

/** 组织部门管理服务 */
class OrganizationServiceImpl implements OrganizationService {
  async CreateOrganization(request: CreateOrganizationRequest): Promise<Empty> {
    return await requestClient.post<Empty>('/orgs', request);
  }

  async DeleteOrganization(request: DeleteOrganizationRequest): Promise<Empty> {
    return await requestClient.delete<Empty>(`/orgs/${request.id}`);
  }

  async GetOrganization(
    request: GetOrganizationRequest,
  ): Promise<Organization> {
    return await requestClient.get<Organization>(`/orgs/${request.id}`);
  }

  async ListOrganization(
    request: PagingRequest,
  ): Promise<ListOrganizationResponse> {
    return await requestClient.get<ListOrganizationResponse>('/orgs', {
      params: request,
    });
  }

  async UpdateOrganization(request: UpdateOrganizationRequest): Promise<Empty> {
    const id = request.org?.id;
    if (request.org !== null) request.org.id = undefined;
    return await requestClient.put<Empty>(`/orgs/${id}`, request);
  }
}

export const defOrganizationService = new OrganizationServiceImpl();
